R/write-traceability-matrix.R

```diff
@@ -43,6 +44,12 @@ requirements and test specifications, are listed in the Requirements Specificati
     filter(!is.na(.data$passed)) %>%
     arrange(.data$StoryId, .data$RequirementId, .data$TestId)
```

Comment:
I haven't done a thorough check of where `RequirementId` is used or manually checked how the story-only path renders, but it looks like this above line should be dropped given the new conditional logic right below.

Reply:
Just to confirm that this is an issue: here’s what I see if I go without the RequirementId and RequirementDescription column (mrgvalidate a560d3b92f4d4ee29b0ca01444520d5e7ab7ad35 and mrgvalprep b28cd6e894b0143a284c9fc8cb10ef14ad35646d): 

```
>
+ df <- create_validation_docs(
+   "Metworx",
+   "21.06",
+   specs = spec,
+   auto_test_dir = "~/scratch/mrgvalidate/auto-tests-v2",
+   roles = roles,
+   man_test_dir = "~/scratch/mrgvalidate/mi-manual-test/metworx-manual-tests-v2",
+ )
+
Finished writing to /Users/KyleM/src/github/metrumresearchgroup/mrgvalidate/requirements-specification.md
  Rendering markdown to docx...
  Finished rendering
Error: arrange() failed at implicit mutate() step.
* Problem with `mutate()` column `..2`.
ℹ `..2 = .data$RequirementId`.
✖ Column `RequirementId` not found in `.data`
Run `rlang::last_error()` to see where the error occurred.
In addition: Warning message:
In check_test_input(bind_rows(results)) : Dropping 65 test(s) with no ID
```